### ISSUE TITLE
Infinite Loop in 1.3.0

### ISSUE THREAD

**Issue by Twipped** [doteric/funpack#16]
I'm guessing this is related to circular dependencies again.

<details><summary>Function</summary>

```js
import { SSMClient, GetParameterCommand } from "@aws-sdk/client-ssm";
import fs from 'fs';

let local;
try {
  const json = fs.readFileSync('./config.json');
  local = JSON.parse(json);
} catch (e) {
  console.error('Could not parse local config: ' + e.message);
  local = {};
}

async function getParameter (Name, WithDecryption = true) {
  const client = new SSMClient();

  const command = new GetParameterCommand({
    Name,
    WithDecryption,
  });

  const { Parameter } = await client.send(command);
  return Parameter;
}

export async function activitypub (event, context) {
  const config = getParameter(local.SSM_PARAMETER);

  const res = {
    statusCode: 200,
    headers: {
      'Content-Type': 'application/json; charset=utf-8',
    },
    body: JSON.stringify({ event, context, local, config }),
  };

  return res;
}
```
</details>

<details><summary>package.json</summary>
```
{
  "name": "@lambdapub/testpoints",
  "main": "testpoints.js",
  "type": "module",
  "scripts": {
    "lint": "eslint .",
    "lint-fix": "eslint --fix .",
    "test": "NODE_ENV=test tap -R dot",
    "build": "funpack"
  },
  "engines": {
    "node": ">=16.13.0"
  },
  "devDependencies": {
    "funpack": "^1.3.0",
  },
  "dependencies": {
    "@aws-sdk/client-ssm": "^3.222.0",
  },
  "funpack": {
    "settings": {
      "outputDir": "../dist",
      "cleanupOutputDir": false,
      "esbuildConfigOverride": {
        "format": "esm",
        "target": "node16",
        "sourcemap": true,
        "splitting": true
      },
      "packageFieldsToCopy": [
        "type"
      ]
    },
    "functions": {
      "activitypub": "./testpoints.js"
    }
  }
}
```
</details>

It drops into a loop of this, over and over:

```
Circular dependency detected! Chain: [
  '../node_modules/@aws-sdk/client-ssm/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/dist-cjs/SSMClient.js',
  '../node_modules/@aws-sdk/client-ssm/dist-cjs/runtimeConfig.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sts/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sts/dist-cjs/STS.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sts/dist-cjs/STSClient.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/credential-provider-node/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/credential-provider-node/dist-cjs/defaultProvider.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/credential-provider-sso/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/credential-provider-sso/dist-cjs/fromSSO.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/credential-provider-sso/dist-cjs/resolveSSOCredentials.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/token-providers/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/token-providers/dist-cjs/fromSso.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/token-providers/dist-cjs/getNewSsoOidcToken.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sso-oidc/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sso-oidc/dist-cjs/SSOOIDCClient.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sso-oidc/dist-cjs/runtimeConfig.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sso-oidc/dist-cjs/runtimeConfig.shared.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/client-sso-oidc/dist-cjs/endpoint/endpointResolver.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/resolveEndpoint.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/index.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateErrorRule.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateExpression.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/callFunction.js',
  '../node_modules/@aws-sdk/client-ssm/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateExpression.js'
]
```

**Comment by doteric** [comment]
Hey @Twipped
I guess the reason is that the `node_modules` are not in the same directory as the `package.json` with the funpack config (or npm package root directory) and rather in the parent directory. This is most likely due to npm workspaces.
I will try to get on this problem in the weekend and find a way to support such style of directories, is that fine? If you want to get it to work asap you could just make sure that the `node_modules` directory is in the same directory as the `package.json` that holds the config of funpack (if that's an option). I will need to create some nice tests for this scenario so it's covered for other cases in the future also.
Big thanks for reporting this problem btw 🙇

**Comment by Twipped** [comment]
That does indeed seem to be the problem. It wouldn't even just let me symlink it, I had to disable workspaces.

**Comment by doteric** [comment]
@Twipped 
I've created a beta version with a potential fix:
https://www.npmjs.com/package/funpack/v/1.3.1-beta-1

Please let me know if it works correctly for you :D

**Comment by doteric** [comment]
@Twipped 
Whenever you have some time then please let me know if the beta version works fine for you as the PR is already prepared, would appreciate it 🙇

**Comment by Twipped** [comment]
Yes! Worked like a charm, thanks!

**Comment by doteric** [comment]
Released in [1.3.1](https://github.com/doteric/funpack/releases/tag/1.3.1) btw 😃
